### ISSUE TITLE
consider switching to preact

### ISSUE THREAD

**Issue by derhuerst** [paritytech/parity#2094]
> [Preact](https://preactjs.com/) is an attempt to recreate the core value proposition of React (or similar libraries like Mithril) using as little code as possible, with first-class support for ES2015. Currently the library is around 3kb (minified & gzipped).
> 
> Preact retains a large amount of compatibility with React, but only the stateless functional components and ES6 Classes interfaces. As one would expect coming from React, Components are simple building blocks for composing a User Interface.

As we use JSX & the React Class syntax for components, theoretically switching shouldn't be difficult.

This plays nicely with #2093 using [`preact-mdl`](https://github.com/developit/preact-mdl/).

**Comment by tomusdrw** [comment]
I was considering this in the past - the only blockers were: `material-ui` and usage of contexts.

**Comment by oliviertassinari** [comment]
@tomusdrw I'm on of the maintainers of `material-ui`. Could you elaborate on the blocker around the usage of contexts?

**Comment by tomusdrw** [comment]
Hi @oliviertassinari, seems that I based my response on outdated knowledge, cause I just checked that Preact was supporting react-compatible contexts back then already.

So if `material-ui` is also compatible, there are no other issues that I know that would block us.

**Comment by oliviertassinari** [comment]
> So if material-ui is also compatible

We have never tried. Bu I definitly think that we should make it compatible if we are not.

**Comment by 5chdn** [comment]
This _optimization_ labelled issue is inactive since February. 

@paritytech/ui-devs  Please, confirm within 7 days this is still _nice to have_. If so, please add an assignee and a milestone, or close otherwise.

**Comment by CraigglesO** [comment]
After playing with Preact, it has some benefits, but still plenty of annoyances as well. I also **don't** think React is the bottleneck _and_ it would take forever to switch.

Lastly some of the biggest cons are:
* Preact doesn't support React PropTypes.
* Preact does not support synthetic event implementation of React. Instead, it relies on browser API. 
* Preact doesn't always call a re-render on certain event changes.
* A lot of other modules would have to change / converted as well (for instance mob-x doesn't work with preact, you need a preact version of mob-x and it's a pain to convert)
* Most importantly the "size" is barebones and you're more than likely going to pull other small support packages to maintain all the functionality. So 3K size is like saying a flight only costs $110, but oh wait, you also have to buy your seats and pay for both overhead luggage space and storage luggage. Oh and tax. So really your flight is $250. LOL, clickbait gotcha.

In short, it would be quiet painful to switch. And React-light has more promise if a change is really desired.
